static/js/containers/PlotContainer.jsx

```diff
@@ -1,12 +1,70 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 
 import Plot from '../components/Plot';
-import { FETCH_SOURCE_PLOT } from '../actions';
-import * as API from '../API';
+import * as Actions from '../actions';
 
-const fetchPlotData = url => (
-  API.GET(url, FETCH_SOURCE_PLOT)
+
+class PlotContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: false,
+      fetchingPlotIDs: []
+    };
+  }
+  componentWillMount() {
+    this.fetchPlotDataIfNotCached();
+  }
+  componentWillReceiveProps() {
+    this.fetchPlotDataIfNotCached();
+  }
```

Comment:
This is not correct; should be replaced by `componentDidMount`.

Reply:
@stefanv we want to check every time the component either a) mounts, or b) receives new props. Can you please explain what you're thinking here?

Reply:
They've now renamed `componentWillReceiveProps` to `UNSAFE_componentWillReceiveProps`, so that's a first indication :)  `componentDidMount` isn't enough in isolation, I imagine, but it looks like `getDerivedStateFromProps` can work in the newest React.  See https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops